### PR TITLE
Fix wrong `floori()` behavior

### LINKED ISSUE #66500: function "floori" returns wrong result for negative numbers
### Godot version

4.0beta1

### System information

macOS Monterey (12.5.1), MacBook Pro, Intel Iris Plus Graphics 1536 MB

### Issue description

The function `floori` does not behave as documented for negative numbers. The documentation states that it is rounding "towards negative infinity", instead it behaves as if it is rounding "towards zero".

See the following list of examples, comparing also with the results on the float-variant, and also the other rounding functions for comparision.

Support for these functions seems to have been added here: https://github.com/godotengine/godot/pull/63361/commits/fdf74410150e6c8750fb251309a1f211d0db03d5

```
floori( 1.3) =  1   floor( 1.3) =  1.000000   (ok)
floori( 0.6) =  0   floor( 0.6) = -0.000000   (ok)
floori(-0.3) =  0   floor(-0.3) = -1.000000   (wrong, should be -1)
floori(-2.3) = -2   floor(-2.3) = -3.000000   (wrong, should be -3)
floori(-2.6) = -2   floor(-2.6) = -3.000000   (wrong, should be -3)
floori(-2.9) = -2   floor(-2.9) = -3.000000   (wrong, should be -3)

roundi( 1.3) =  1   round( 1.3) =  1.000000   (ok)
roundi( 0.6) =  1   round( 0.6) =  1.000000   (ok)
roundi(-0.3) =  0   round(-0.3) = -0.000000   (ok)
roundi(-2.3) = -2   round(-2.3) = -2.000000   (ok)
roundi(-2.6) = -3   round(-2.6) = -3.000000   (ok)
roundi(-2.9) = -3   round(-2.9) = -3.000000   (ok)

 ceili( 1.3) =  2    ceil( 1.3) =  2.000000   (ok)
 ceili( 0.6) =  1    ceil( 0.6) =  1.000000   (ok)
 ceili(-0.3) =  0    ceil(-0.3) = -0.000000   (ok)
 ceili(-2.3) = -2    ceil(-2.3) = -2.000000   (ok)
 ceili(-2.6) = -2    ceil(-2.6) = -2.000000   (ok)
 ceili(-2.9) = -2    ceil(-2.9) = -2.000000   (ok)
```

### Steps to reproduce

- simply use `floori` with a negative number

### Minimal reproduction project

`assert(floori(-1.6) == -2)`

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #66500

### CODE REVIEW

**Comment by MewPurPur** [comment]
Shouldn't this be cast to an int? ceili does it.

**Review by akien-mga — APPROVED** [review]

File: core/variant/variant_utility.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ struct VariantUtilityFunctions {
 	}
 
 	static inline int floori(double x) {
-		return int(x);
+		return int(Math::floor(x));
```
[the remark this answers is not in the document]
Comment (KoBeWi) [reply core/variant/variant_utility.cpp:132]: All other methods are using `int` too. Should I change it all in this PR?

[the remark this answers is not in the document]
Comment (akien-mga) [reply core/variant/variant_utility.cpp:132]: Yeah I guess they can all be fixed at the same time.

[the remark this answers is not in the document]
Comment (Clonkex) [reply core/variant/variant_utility.cpp:132]: > All other methods are using `int` too. Should I change it all in this PR?

I know nothing, being new to contributing on Github, but if it were me, I would be inclined to create a separate PR. This PR is for fixing wrong `floori` behaviour. It makes sense to me that you should create a new PR for fixing `floori`, `ceili`, `roundi` etc. to be properly 64bit.

[the remark this answers is not in the document]
Comment (akien-mga) [reply core/variant/variant_utility.cpp:132]: Fair enough, let's merge this fix first.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**